src/agents/gc/OSConfig_PrototypeResource.c

```diff
@@ -0,0 +1,658 @@
+                    miResult = MI_Instance_SetElement(resultResourceObject, MI_T(allParameters[i].name), &miValue, MI_STRING, 0);
+                }
+                else
+                {
+                    OsConfigLogError(GetLog(), "GetTargetResource: no string value for %s", allParameters[i].name);
+                    miResult = MI_RESULT_FAILED;
+                }
+                break;
+
+            case MI_BOOLEAN:
+                miValue.boolean = (MI_Boolean)(allParameters[i].booleanValue);
+                miResult = MI_Instance_SetElement(resultResourceObject, MI_T(allParameters[i].name), &miValue, MI_BOOLEAN, 0);
+                break;
+
+            case MI_UINT32:
+            default:
+                miValue.uint32 = (MI_Uint32)(allParameters[i].integerValue);
+                miResult = MI_Instance_SetElement(resultResourceObject, MI_T(allParameters[i].name), &miValue, MI_UINT32, 0);
+        }
+
+        if (MI_RESULT_OK != miResult)
+        {
+            OsConfigLogError(GetLog(), "GetTargetResource: MI_Instance_SetElement(%s) failed with %d", allParameters[i].name, miResult);
+        }
+
+        memset(&miValue, 0, sizeof(miValue));
+    }
+    
+    // Set the created output resource instance as the output resource in the GetTargetResource instance
+    if (MI_RESULT_OK != (miResult = MI_Instance_SetElement(&get_result_object.__instance, MI_T("OutputResource"), &miValueResource, MI_INSTANCE, 0)))
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: MI_Instance_SetElement(OutputResource) failed with %d", miResult);
+        goto Exit;
+    }
+    else
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: OSConfig_PrototypeResource_GetTargetResource_Post failed with %d", miResult);
+    }
+        
+Exit:
+    if (MI_RESULT_OK != miResult)
+    {
+        MI_Context_PostError(context, miResult, MI_RESULT_TYPE_MI, 0);
+    }
+
+    // Clean up the Result MI value instance if needed
+    if ((NULL != miValueResult.instance) && (MI_RESULT_OK != MI_Instance_Delete(miValueResult.instance)))
+    {
+        OsConfigLogInfo(GetLog(), "GetTargetResource: MI_Instance_Delete(miValueResult) failed");
+    }
+
+    // Clean up the output resource instance
+    if ((NULL != resultResourceObject) && (MI_RESULT_OK != MI_Instance_Delete(resultResourceObject)))
+    {
+        OsConfigLogInfo(GetLog(), "GetTargetResource: MI_Instance_Delete(resultResourceObject) failed");
+    }
+
+    // Clean up the GetTargetResource instance
+    if (MI_RESULT_OK != (miResult == OSConfig_PrototypeResource_GetTargetResource_Destruct(&get_result_object)))
+    {
+        OsConfigLogError(GetLog(), "GetTargetResource: OSConfig_PrototypeResource_GetTargetResource_Destruct failed with %d", miResult);
+    }
+
+    // Post MI result back to MI to finish
+    OsConfigLogInfo(GetLog(), "GetTargetResource: complete");
+    MI_Context_PostResult(context, miResult);

```

Comment:
In the sample for **GET** only, `MI_Context_PostResult` is not called if `MI_Context_PostError` was called earlier. Perhaps check `result == MI_RESULT_OK` first.

Reply:
MI_ContextPostResult is not intended for just MI_RESULT_OK. My understating is that can also return any of the following:

 *      MI_RESULT_ACCESS_DENIED
 *      MI_RESULT_INVALID_NAMESPACE
 *      MI_RESULT_INVALID_PARAMETER
 *      MI_RESULT_INVALID_CLASS
 *      MI_RESULT_NOT_FOUND
 *      MI_RESULT_FAILED